src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -425,13 +425,18 @@ private Territory getRetreatInternal(final String message, final Collection<Terr
       // if you have eliminated the impossible, whatever remains, no matter
       // how improbable, must be the truth
       // retreat
-      final RetreatComponent comp = new RetreatComponent(possible);
-      final int option = JOptionPane.showConfirmDialog(BattleDisplay.this, comp, message,
-          JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE, null);
-      if (option == JOptionPane.OK_OPTION) {
-        if (comp.getSelection() != null) {
-          retreatTo[0] = comp.getSelection();
-          latch.countDown();
+      if (possible.size() == 1) {
+        retreatTo[0] = possible.iterator().next();
+        latch.countDown();
```

Comment:
I don't think I was clear. Imagine I have unit A that attacks territory 1 and unit B that attacks territory 2 (adjacent to territory 1). I fight the battle for territory 1 and win. Then I'm fighting the battle in territory 2 and decide I might want to retreat and I forget that I didn't attack from territory 1. So I'm thinking I can retreat there when in reality I can't (many times users forget which territories they attacked from and therefore are allowed to retreat to).